internal/controller/promotions/promotions.go

```diff
@@ -544,6 +546,10 @@ func (r *reconciler) buildTargetFreightCollection(
 	lastPromo := stage.Status.LastPromotion
 	if lastPromo != nil {
 		for _, req := range stage.Spec.RequestedFreight {
+			if lastPromo.Status == nil || lastPromo.Status.FreightCollection == nil || lastPromo.Status.FreightCollection.Freight == nil {
```

Comment:
```suggestion
			if lastPromo.Status == nil ||
				lastPromo.Status.FreightCollection == nil ||
				lastPromo.Status.FreightCollection.Freight == nil {
```

to please the linter.